zad5.py

```diff
@@ -1,49 +1,47 @@
+from math import log10
 from random import randint
-end = None
 
 
 def is_prime(n):
     if n < 2:
         return False
-    if n == 2:
+    elif n == 2:
         return True
+    elif n % 2 == 0:
+        return False
 
-    x = 2
-    while x*x <= n:
-        if n % x == 0:
+    div = 3
+    while div * div <= n:
+        if n % div == 0:
             return False
-
-        x += 2
-    end
-
+        div += 2
     return True
 
 
-def can_cut(tab):
-    a = tab[0] * 2 + tab[1]
-
-    b = 0
-    for i in tab[2:]:
-        b <<= 1
-        b |= i
-    end
+def bin_to_dec(num):
+    a = 0
+    i = 0
+    while num > 0:
+        num, rem = divmod(num, 10)
+        a += rem * (2 ** i)
+        i += 1
+    return a
 
-    for i in range(2, len(tab) - 2):
-        # print("a =", bin(a), "b =", bin(b))
 
-        if is_prime(a) and is_prime(b):
-            print(a, b)
+def split(tab, pos=0, num=0):
+    # print(num)
+    if not pos >= len(tab):
+        a = split(tab, pos + 1, num * 10 + tab[pos])
+        if a:
             return True
-
-        a <<= 1
-        a |= tab[i]
-
-        b -= tab[i] << (len(tab) - i - 1)
-    end
-
-    return False
+        b = False
+        if is_prime(bin_to_dec(num)):
+            b = split(tab, pos + 1, tab[pos])
+        return b
+    if log10(bin_to_dec(num) + 1) != len(tab):
```

Comment:
ej bo to zwraca float
`log10(123 + 1) = 2.093421685162235`
a chyba nie o to chodzi

Reply:
jest git